qctrlopencontrols/dynamic_decoupling_sequences/driven_controls.py

```diff
@@ -270,6 +273,18 @@ def convert_dds_to_driven_control(
                                   extras={'deduced_pulse_start_timing': pulse_start_ends[:, 0],
                                           'deduced_pulse_end_timing': pulse_start_ends[:, 1]})
 
+    # check if the minimum_segment_duration is respected in the gaps between the pulses
+    if (np.any(pulse_start_ends[1:, 0]-pulse_start_ends[:-1, 1] < minimum_segment_duration)):
```

Comment:
I think that's a fair point, I've re-written the error to allow some tolerance for the floating point imprecisions. (It would be nice if we could have `less_or_close` and `greater_or_close` as functions we can import from somewhere so we don't have to effectively re-write them in every file.)

I think you're right that the lack of floating point tolerance is what made Luigi's test fail yesterday. I think it might also be worth rewriting the tests above so that the error messages are a little bit more helpful as well (at the moment they don't say which of the four conditions was violated). I feel this is a bit beyond the scope of this PR, so maybe I can do it next and add it to the same release of Open Controls as this